Geant4.10.04 Issue 65 Fix

cnatzke opened the pull request:
I changed the naming convention of the BGO's to include detector number. Previously they were named based on an iterator so placing non-sequential detectors (e.g. 1, 3, 11, anad 14) caused a discrepancy in BGO compared to the clovers; now they follow the same method as the clovers.

VinzenzBildstein commented:
- Why did you change the name of the logical volume from rightSuppressorShellLog to rightSuppressorShell (and similar for the other logical suppressor volumes)?
- Why are only the left and right suppressors using the detector number in their name and not the other suppressors?
- Why do the back quarter suppressors not get at system ID anymore?
- Is it not necessary to change the stepping action as well? As far as I can see the function SetDetAndCryNumberForGriffinComponent assumes the detector number to be in characters 3-5 (counting from 0) which isn't true anymore?

cnatzke commented:
> Why did you change the name of the logical volume from rightSuppressorShellLog to rightSuppressorShell (and similar for the other logical suppressor volumes)?

I changed them to match the pattern of the Germanium crystals, e.g. rightSuppressorShell_<detector_num>_log

> Why are only the left and right suppressors using the detector number in their name and not the other suppressors?
Why do the back quarter suppressors not get at system ID anymore?

I deleted too many things and did not double check my work. I have fixed that in the my current branches. 

>Is it not necessary to change the stepping action as well? As far as I can see the function SetDetAndCryNumberForGriffinComponent assumes the detector number to be in characters 3-5 (counting from 0) which isn't true anymore?

Where are you finding this? I don't see a function named "SetDetAndCryNumberForGriffinComponent" in SteppingAction

VinzenzBildstein commented:
Hmm, not sure where I was seeing that, as it isn't there anymore, but I though the diff showed that one of the names was hard-coded to be "rightSuppressorShell" without any _<detector_num>_log. Maybe I just imagined that. Anyway that is fixed as well as the other two points I think. Not sure where I found the SetDetAndCryNumberForGriffinComponent function, might have been a different branch of the code. Looks good to me now, I'll merge it.
